### PR TITLE
fix(tests): eliminate budget tracker race condition in agent unit tests

### LINKED ISSUE #275: Flaky test: test_status_returns_last_cycle_when_idle - budget tracker race condition
## Summary

The test `test_status_returns_last_cycle_when_idle` in `tests/unit/agent_tests/test_agent_service_new.py` is flaky due to a race condition with the budget tracker's file operations during parallel test execution.

## Error

```
FAILED tests/unit/agent_tests/test_agent_service_new.py::TestAgentServiceGetStatus::test_status_returns_last_cycle_when_idle 
- json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

The error occurs in `ktrdr/agents/budget.py:66` when `_load_today()` tries to load a JSON file that is empty.

## Root Cause

This is a test isolation issue. The budget tracker uses a shared file path, and when tests run in parallel:
1. One test creates/truncates the budget file
2. Another test tries to read it while it's empty
3. `json.loads()` fails on empty string

## Current Mitigation

The test has been marked as `@pytest.mark.skip` to unblock CI.

## Proposed Fix

Options:
1. **Mock the budget tracker** in this test to avoid file operations
2. **Use test-specific file paths** for the budget tracker via fixture
3. **Add file locking** to the budget tracker (overkill for this use case)

Option 1 (mocking) is likely the cleanest approach for unit tests.

## Files Affected

- `tests/unit/agent_tests/test_agent_service_new.py` - the flaky test
- `ktrdr/agents/budget.py` - the budget tracker with file operations

### PR DESCRIPTION
**Pull request by kpiteira**
## Summary

- Fixed cross-process race condition where pytest-xdist workers shared a real budget JSON file, causing intermittent `JSONDecodeError` in CI
- Added `cleanup_coordinator` async fixture to cancel leaked background tasks from `trigger()` tests (4 test classes across 3 files)
- Added `mock_budget` fixture to prevent filesystem access in `get_status()` reader tests (3 test classes)
- Unskipped `test_status_returns_last_cycle_when_idle` which was marked flaky in #275

## Root Cause

Three interacting bugs:
1. **Leaked async tasks**: `service.trigger()` spawns background coordinator tasks never cancelled after tests
2. **Incomplete mocking**: Budget mocked at `agent_service` import but not at `research_worker` import — background tasks called real `record_spend()`
3. **Missing mock in readers**: `get_status()` tests read the real budget file that writers were racing on

## Test plan

- [x] `make test-unit` passes (4320 passed, 5 skipped — was 4319/6)
- [x] `make quality` passes
- [x] Parallel xdist stress test: `pytest tests/unit/agent_tests/ -v -n 4 --no-cov` — 545 passed, 0 warnings
- [x] No `PytestDeprecationWarning` for async fixtures

Closes #275

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by kpiteira** [comment]
@claude review

File: tests/unit/agent_tests/test_agent_service_new.py [hunk 1]
```diff
@@ -127,6 +128,18 @@ def use_stub_workers(self, monkeypatch):
...
+    async def cleanup_coordinator(self):
+        """Cancel leaked background coordinator tasks after each test."""
+        yield
+        for task in asyncio.all_tasks():
+            if task is not asyncio.current_task() and not task.done():
+                task.cancel()
+                try:
+                    await task
+                except (asyncio.CancelledError, Exception):
+                    pass
```
[the remark this answers is not in the document]
Comment (kpiteira) [reply tests/unit/agent_tests/test_agent_service_new.py]: Keeping the current approach. In pytest-asyncio strict mode, each test gets its own event loop scope — the only non-current, non-done tasks at teardown are leaked application tasks (coordinator coroutines from `trigger()`). There are no pytest infrastructure tasks in the async task set.

Tracking specific coordinator tasks would require changing `AgentService` to expose task references, which adds production complexity solely for test cleanup. The current approach is validated by 4320 passing tests with 0 warnings under parallel xdist execution.

Added an inline comment to make the intent clear.
